draft-ietf-quic-transport.md

```diff
@@ -1461,127 +1461,126 @@ connection if the integrity check fails with a PROTOCOL_VIOLATION error code.
+period, but not receiving a PATH_RESPONSE may be indicative of a PMTU detection
+failure.
+
+A client may receive a PATH challenge from a server that is validating the
+client's ownership of an address. A client responds to this frame by echoing the
+data from the PATH_CHALLENGE frame in a PATH_RESPONSE frame.  This response MAY
+be bundled with other frames and does not require additional padding.
+
+
+### Server Behavior {{migration-server}}
+
+When a server receives a PATH_CHALLENGE frame from a client, it MUST respond by
+copying the included data in a PATH_RESPONSE frame, and send it in a packet
+padded to the size of the packet carrying the PATH_CHALLENGE. This PATH_RESPONSE
+MUST be sent to the address that the PATH_CHALLENGE was received from.
+
+The server need not retransmit its PATH_RESPONSE, since an interested client is
+expected to send more PATH_CHALLENGE frames when a response is not received for
+a prior one.
+
+Before sending any other packets to a client address that is not yet considered
+valid, the server MUST send a PATH_CHALLENGE frame with its own validation data
+to verify the client's ownership of this address. The server's PATH_CHALLENGE
+MAY be bundled with other frames in a packet that does not need to be padded.
+
+When the server receives a PATH_RESPONSE from the client and verifies that the
+included data matches that sent by the server in a previous PATH_CHALLENGE, the
+server considers the client's address to which the corresponding PATH_CHALLENGE
+was sent as valid.
+
+A server MAY send additional PATH_CHALLENGE frames to handle packet loss,
+subject to the congestion controller's limits. The server MUST use fresh random
+data in each PATH_CHALLENGE frame so that it can associate the client's response
+with the causative PATH_CHALLENGE.
+
+Until a client's address is deemed valid, the server MUST limit the rate at
+which it sends data to the address. The server SHOULD NOT send more than an
+initial window's worth of bytes (as defined in {{QUIC-RECOVERY}}}) per roundtrip
+time.  In the absence of this limit, the server risks being used for launching a
+denial of service attack against an unsuspecting victim.
+
+A server MAY skip address validation for a client address that it considers to
+be already valid.
+
+After validating a client address, the server MAY send an updated address
+validation token to the client ({{address-validation}}).
+
+When validation of an address fails, the server SHOULD revert back to using the
+last validated client address, absent which the server MUST close the
 connection.
 
+When a server receives a packet containing a frame other than PATH_CHALLENGE,
+PATH_RESPONSE, or PADDING from a client address with a packet number that is the
+largest seen thus far on the connection, the server commits to this client
+address. The server MUST send all subsequent packets, with the exception of
+those containing PATH_RESPONSE and PATH_CHALLENGE frames, to this address.
```

Comment:
Do we want to explicitly mention the converse of this statement (to make sure implementations do it right), that just sending PATH_CHALLENGE/PATH_RESPONSE/PADDING only probes a path without changing where the server sends data back to the client?

Reply:
Done.

Reply:
Looks perfect